### PR TITLE
docs: re-organize top level model definitions docs

### PR DESCRIPTION
**Pull request by yoavz**
## Description
Previously, the journey of a user from the top-level of the docs was focused on the Trial API. This re-organization makes it clearer that there are two distinct experiment APIs and provides the user context about which path to take. 

## Test Plan
Proofread. 

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/topic-guides/model-definitions/index.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+SDK supports the Native API for TF Keras and Estimators.
+
+To read more about the Trial API, see:
+
+-  :ref:`model-definitions_trial-api`
+-  :ref:`pytorch-trial`
+-  :ref:`tf-keras-trial`
+-  :ref:`estimator-trial`
+
+To read more about the Native API see:
```
Comment (davetroiano) [inline docs/topic-guides/model-definitions/index.txt:27]: for consistency with line 20, add comma after `API`

File: docs/index.txt [hunk 2]
```diff
@@ -45,13 +45,25 @@ Each user should also :ref:`install the Determined command-line tools <install-c
 
 Next Steps
 """"""""""
+
 To get started using Determined, check out the :ref:`quick-start`.
 
-Next, follow the tutorial for your preferred deep learning framework:
+Determined has two Experiment APIs: Trial and Native.  With the :ref:`model-definitions_trial-api`, developers expose the components of a model definition to Determined’s training loop implementation.  With the :ref:`model-definitions_native-api` (TensorFlow Keras and Estimators), developers instrument training scripts adhering to these frameworks' native patterns including their training loops.
```
Comment (justin-determined-ai) [inline docs/index.txt:51]: blocking: I would list these in the same order you describe them in the next paragraphs, and I'd put Native in front of Trial.

File: docs/index.txt [hunk 3]
```diff
@@ -45,13 +45,25 @@ Each user should also :ref:`install the Determined command-line tools <install-c
...
+
+- :ref:`tutorials_native-api`
+
+If you're using PyTorch or want a finer-grained level of control over the
+training loop, follow our Trial API tutorials in your preferred framework:
 
 - :ref:`pytorch-mnist-tutorial`
 - :ref:`tf-mnist-tutorial`
 
+The command line interface and Python SDK support the Trial API.  The Python
```
Comment (justin-determined-ai) [inline docs/index.txt:64]: blocking: As above, describe the two APIs in the same order every time they're paired here.

File: docs/release-notes.txt [hunk 4]
```diff
@@ -29,7 +29,7 @@ Version 0.12.4
 
 - Add a topic guide explaining Determined's approach to :ref:`elastic-infra-index`.
 
-- Add a topic guide explaining the :ref:`native-api`.
+- Add a topic guide explaining the :ref:`model-definitions_native-api`.
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:32]: blocking: I assume find-and-replace gone awry? Don't need to update release notes :)

File: docs/topic-guides/index.txt [hunk 5]
```diff
@@ -26,8 +26,12 @@ Topic guides discuss different concepts and topics about Determined and deep lea
 - :ref:`dynamic-agents-aws`
 - :ref:`dynamic-agents-gcp`
 
-**Best Practices**
-  - :ref:`best-practices-model-definitions`
+**Model Definitions**
+
+- :ref:`model-definitions`
+- :ref:`model-definitions_trial-api`
```
Comment (justin-determined-ai) [inline docs/topic-guides/index.txt:32]: non-blocking: I'd personally put Native above Trial.

File: docs/topic-guides/index.txt [hunk 6]
```diff
@@ -36,15 +40,16 @@ The full list of our topic guides can be found below:
...
   effective-distributed-training
   experiment-lifecycle
   checkpoints
   hp-tuning-defined
   hp-tuning-det/index
   jupyter-notebooks
-  model-definitions
-  native-api
+  model-definitions/index
+  model-definitions/trial-api
```
Comment (justin-determined-ai) [inline docs/topic-guides/index.txt:50]: non-blocking: as above

File: docs/topic-guides/model-definitions/index.txt [hunk 7]
```diff
@@ -0,0 +1,43 @@
+.. _model-definitions:
+
+Model Definitions
```
Comment (justin-determined-ai) [inline docs/topic-guides/model-definitions/index.txt:3]: blocking: Now that I'm getting to here, it's feeling slightly redundant with the index. I'd err on the side of doing a much shorter intro in index and linking here instead.

File: docs/index.txt (hunk 2 above)
Comment (justin-determined-ai) [reply docs/index.txt:51]: Also, not clear to me if this should be done _after_ quick-start, or is this if you don't want to use quick start? Would this go at the end of the quick-start guide too?

File: docs/topic-guides/model-definitions/native-api.txt [hunk 8]
```diff
@@ -1,4 +1,4 @@
-.. _native-api:
+.. _model-definitions_native-api:
 
 Native API (experimental)
```
Comment (justin-determined-ai) [inline docs/topic-guides/model-definitions/native-api.txt:3]: blocking: Remove "(experimental)"?

File: docs/topic-guides/model-definitions/trial-api.txt [hunk 9]
```diff
@@ -1,19 +1,14 @@
...
+.. _model-definitions_trial-api:
 
-Model Definitions
-=================
-
-The model definition is the interface between Determined and the user's
-application framework (e.g., Keras, TensorFlow), in terms of loading
-training data, describing a model architecture, and specifying the
-underlying iterative optimization training algorithms.
+Trial API
```
[the remark this answers is not in the document]
Comment (yoavz) [reply docs/topic-guides/model-definitions/trial-api.txt:3]: Did a second pass making most of these changes.

File: docs/topic-guides/model-definitions/native-api.txt (hunk 8 above)
[the remark this answers is not in the document]
Comment (yoavz) [reply docs/topic-guides/model-definitions/native-api.txt:3]: removed

Comment (yoavz) [reply docs/topic-guides/model-definitions/native-api.txt]: removed

File: docs/index.txt (hunk 2 above)
Comment (justin-determined-ai) [reply docs/index.txt]: Okay, I won't block, but I think making it a bit more narrative will help. "We recommend you follow our [Quick Start](url) guide if you're new to Determined. Next, learn more about our different APIs and how to interact with them." (or something like this)

File: docs/topic-guides/model-definitions/index.txt (hunk 7 above)
Comment (justin-determined-ai) [reply docs/topic-guides/model-definitions/index.txt:3]: I guess I was envisioning that you would say "First start with Quick Start, then read Model Definitions for a deeper understanding" or something. But this is mostly my gut feeling so I won't block here.

File: docs/release-notes.txt (hunk 4 above)
Comment (justin-determined-ai) [reply docs/release-notes.txt:32]: Oh oops, duh. Makes sense!

File: docs/index.txt (hunk 2 above)
Comment (yoavz) [reply docs/index.txt]: done

**Pull request merged**